runtime/devnet/src/tests/mod.rs

```diff
@@ -0,0 +1,262 @@
+		assert_eq!(encoded, vec![3, 52, 18, 0, 0, 0]);
+		assert_eq!(
+			decoded,
+			DispatchError::Module(ModuleError { index: 52, error: [18, 0, 0, 0], message: None })
+		);
+		println!("Encoded Module Error: {:?}", encoded);
+
+		// Example DispatchError::Token
+		let error = DispatchError::Token(TokenError::UnknownAsset);
+		let encoded = error.encode();
+		assert_eq!(encoded, vec![7, 4]);
+		println!("Encoded Token Error: {:?}", encoded);
+
+		// Example DispatchError::Arithmetic
+		let error = DispatchError::Arithmetic(ArithmeticError::Overflow);
+		let encoded = error.encode();
+		assert_eq!(encoded, vec![8, 1]);
+		println!("Encoded Arithmetic Error: {:?}", encoded);
+	});
+}
+
+#[test]
+fn encoding_of_enum() {
+	use codec::{Decode, Encode};
+
+	// Comprehensive enum with all different type of variants.
+	#[derive(Debug, PartialEq, Encode, Decode)]
+	enum ComprehensiveEnum {
+		SimpleVariant,
+		DataVariant(u8),
+		NamedFields { w: u8 },
+		NestedEnum(InnerEnum),
+		OptionVariant(Option<u8>),
+		VecVariant(Vec<u8>),
+		TupleVariant(u8, u8),
+		NestedStructVariant(NestedStruct),
+		NestedEnumStructVariant(NestedEnumStruct),
+	}
+
+	#[derive(Debug, PartialEq, Encode, Decode)]
+	enum InnerEnum {
+		A,
+		B { inner_data: u8 },
+		C(u8),
+	}
+
+	#[derive(Debug, PartialEq, Encode, Decode)]
+	struct NestedStruct {
+		x: u8,
+		y: u8,
+	}
+
+	#[derive(Debug, PartialEq, Encode, Decode)]
+	struct NestedEnumStruct {
+		inner_enum: InnerEnum,
+	}
+
+	// Creating each possible variant for an enum.
+	let enum_simple = ComprehensiveEnum::SimpleVariant;
+	let enum_data = ComprehensiveEnum::DataVariant(42);
+	let enum_named = ComprehensiveEnum::NamedFields { w: 42 };
+	let enum_nested = ComprehensiveEnum::NestedEnum(InnerEnum::B { inner_data: 42 });
+	let enum_option = ComprehensiveEnum::OptionVariant(Some(42));
+	let enum_vec = ComprehensiveEnum::VecVariant(vec![1, 2, 3, 4, 5]);
+	let enum_tuple = ComprehensiveEnum::TupleVariant(42, 42);
+	let enum_nested_struct = ComprehensiveEnum::NestedStructVariant(NestedStruct { x: 42, y: 42 });
+	let enum_nested_enum_struct = ComprehensiveEnum::NestedEnumStructVariant(NestedEnumStruct {
+		inner_enum: InnerEnum::C(42),
+	});
+
+	// Encode and print each variant individually to see their encoded values.
+	println!("{:?} -> {:?}", enum_simple, enum_simple.encode());
+	println!("{:?} -> {:?}", enum_data, enum_data.encode());
+	println!("{:?} -> {:?}", enum_named, enum_named.encode());
+	println!("{:?} -> {:?}", enum_nested, enum_nested.encode());
+	println!("{:?} -> {:?}", enum_option, enum_option.encode());
+	println!("{:?} -> {:?}", enum_vec, enum_vec.encode());
+	println!("{:?} -> {:?}", enum_tuple, enum_tuple.encode());
+	println!("{:?} -> {:?}", enum_nested_struct, enum_nested_struct.encode());
+	println!("{:?} -> {:?}", enum_nested_enum_struct, enum_nested_enum_struct.encode());
+}
+
+#[test]
+fn dispatch_error_to_status_code_to_pop_api_error_works() {
```

Comment:
These are in the wrong place. The pop-api crate should take a dev-dependency on the devnet runtime instead and the tests should be located there.

This is creating a cycle of confusion: pop-api < runtime < pop-api
It should just be: runtime < pop-api

Reply:
Resolved